src/backend/pipeline/tuplebuf.c

```diff
@@ -614,7 +614,7 @@ TupleBufferTryWait(TupleBufferReader *reader)
 	SpinLockRelease(&buf->mutex);
 
 	if (should_wait)
-		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 0);
+		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 1);
```

Comment:
This fix might not be too great then because this means that our background processes will never go to sleep.